Helpers/TaskHelpers.cs

```diff
@@ -1,28 +1,65 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using CRCTray.Communication;
 
 namespace CRCTray.Helpers
 {
     internal class TaskHelpers
     {
-        public static async Task<T> TryTask<T>(Func<T> function)
+        public static async Task<T> TryTask<T>(Func<T> function, bool retryable = false)
         {
             try
             {
                 return await Task.Run(function);
             }
             catch (AggregateException ae)
             {
-                handleAggregate(ae);
+                handleAggregate(ae, retryable: retryable);
             }
 
             return default;
         }
 
+        public static async Task<T> TryTaskWithRetry<T>(Func<T> function, int Count = 3, int backOff = 2000)
+        {
+            var RetryCount = 0;
+
+            for(; ;)
+            {
+                if(RetryCount > Count)
+                {
+                    throw new RetryableTaskFailedException($"Re-tried task {Count} times");
+                }
+
+                try
+                {
+                    return await TryTask(function, retryable: true);

```

Comment:
Reused `TryTask` but had to introduce the `retryable` parameter, with this we skip handling `TimeoutException` and `TaskCanceledException` in `handleAggregate